File: leetcode/js/count-symmetric-integers.js
/**
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var countSymmetricIntegers = function (low, high) {
  let result = 0;

  for (let num = low; num <= high; num++) {
    const strNum = String(num);

    if (strNum.length % 2 !== 0) {
      continue;
    }

    const sum1 = [...strNum.slice(0, strNum.length / 2)].reduce(
      (acc, cur) => acc + Number(cur),
      0
    );
    const sum2 = [...strNum.slice(strNum.length / 2)].reduce(
      (acc, cur) => acc + Number(cur),
      0
    );

    if (sum1 === sum2) {
      result++;
    }
  }

  return result;
};

File: leetcode/python/count-symmetric-integers.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        result = 0

        for num in range(low, high + 1):
            str_num = str(num)
            num_len = len(str_num)

            if num_len % 2 == 0:
                mid = num_len // 2

                first_sum = sum(int(c) for c in str_num[:mid])
                last_sum = sum(int(c) for c in str_num[mid:])

                result += 1 if first_sum == last_sum else 0
        
        return result